pkg/ipamd/ipamd.go

```diff
@@ -529,7 +529,11 @@ func (c *IPAMContext) nodeInit() error {
 
 	// On node init, check if datastore pool needs to be increased. If so, attach CIDRs from existing ENIs and attach new ENIs.
 	if !c.disableENIProvisioning && c.isDatastorePoolTooLow() {
-		if err := c.increaseDatastorePool(ctx); err != nil {
```

Comment:
nit: I am realizing now that `increaseDatastorePool` is an expensive function, so the better check is to loop on `c.isDatastorePoolTooLow` returning true. As long as we are supposed to increase the datastore pool, we should do it.

Now, the question to think about is whether we would do this only during the main IP pool loop, or during initialization as well

Reply:
ack